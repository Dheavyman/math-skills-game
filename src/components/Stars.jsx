
import React from 'react';

/**
 * Stars component
 *
 * @returns {object} React element
 */
const Stars = () => {
  const numberOfStars = 1 + Math.floor(Math.random() * 9);

  return (
    <div className="col-5">
      {Array.from({ length: numberOfStars }, (_, index) => index + 1).map(
        (star, index) => (
          <i key={index.toString()} className="fa fa-star" />
        )
      )}
    </div>
  );
};

export default Stars;
