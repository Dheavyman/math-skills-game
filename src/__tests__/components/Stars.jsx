import React from 'react';
import TestRenderer from 'react-test-renderer';

import Stars from '../../components/Stars';

describe('Stars component', () => {
  const props = {
    numberOfStars: 4
  };
  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<Stars {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
