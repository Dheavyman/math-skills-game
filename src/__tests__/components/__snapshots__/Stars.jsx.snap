// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars component should render component successfully 1`] = `
<div
  className="col-5"
>
  <i
    className="fa fa-star"
  />
  <i
    className="fa fa-star"
  />
  <i
    className="fa fa-star"
  />
  <i
    className="fa fa-star"
  />
</div>
`;
